contrib/babelfishpg_tsql/src/forjson.c

```diff
@@ -55,6 +57,31 @@ SPI_sql_row_to_json_path(uint64 rownum, StringInfo result, bool include_null_val
 	const char  *sep="";
 	bool 		isnull;
 
+	/*
+	* TODO: encode binary/varbinary/image data values using base64 encoding.
+	* Also, pg_b64_encode function might be useful.
+	* For now report an ERROR if any attribute is binary data type since base64
+	* encoding is not implemented yet.
+	* Also Exact numeric's datatype money or smallmoney is not supported.
+	*/
+	for (i = 1; i <= SPI_tuptable->tupdesc->natts; i++)
+	{
+		char* typename = SPI_gettype(SPI_tuptable->tupdesc, i);
+		if (strcmp(typename, "binary") == 0 ||
+			strcmp(typename, "varbinary") == 0 ||
+			strcmp(typename, "image") == 0 ||
+			strcmp(typename, "timestamp") == 0)
+			ereport(ERROR,
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("option binary base64 is not supported")));
```

Comment:
1. Is this a proper error message for `image`, `timestamp` and `rowversion`?
2. Please add test cases for all unsupported data types.

Reply:
Adding [link](https://learn.microsoft.com/en-us/sql/relational-databases/json/how-for-json-converts-sql-server-data-types-to-json-data-types-sql-server?view=sql-server-ver16#:~:text=The%20FOR%20JSON%20clause%20uses%20the%20following%20rules%20to%20convert%20SQL%20Server%20data%20types%20to%20JSON%20types%20in%20the%20JSON%20output) for reference.